go/cloudtrail/cloudtrailOps.go

```diff
@@ -0,0 +1,476 @@
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+   This file is licensed under the Apache License, Version 2.0 (the "License").
+   You may not use this file except in compliance with the License. A copy of
+   the License is located at
+
+    http://aws.amazon.com/apache2.0/
+
+   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations under the License.
+*/
+
+package main
+
+import (
+    "encoding/json"
+    "fmt"
+    "io/ioutil"
+    "os"
+    "strings"
+    "time"
+
+    "github.com/aws/aws-sdk-go/aws"
+    "github.com/aws/aws-sdk-go/aws/session"
+    "github.com/aws/aws-sdk-go/service/cloudtrail"
+    "github.com/aws/aws-sdk-go/service/s3"
+    "github.com/aws/aws-sdk-go/service/sts"
+)
+
+// CreateTrail creates a trail that gets event activity from a bucket
+// Inputs:
+//     trailName is the name of the trail
+//     bucketName is the name of the bucket
+// Output:
+//     If success, nil
+//     Otherwise, an error from the call to sts.GetCallerIdentity, json.Marshal, s3.PutBucketPolicy, or CreateTrail
+// snippet-start:[cloudtrail.go.create_trail]
+func CreateTrail(sess *session.Session, trailName string, bucketName string) error {
+    // snippet-start:[cloudtrail.go.create_trail_policy]
+    stsSvc := sts.New(sess)
+    input := &sts.GetCallerIdentityInput{}
+
+    result, err := stsSvc.GetCallerIdentity(input)
+    if err != nil {
+        return err
+    }
+
+    accountID := aws.StringValue(result.Account)
+
+    // Create policy to allow trail to write to bucket
```

Comment:
...allow **the** trail **...** to **the** bucket

Reply:
The comment has changed.